sourcegraph/sourcegraph
# notebooks: improved sharing UI

Fixes #30392

## Demo

https://user-images.githubusercontent.com/6417322/153391497-156842df-eac5-409c-abe4-d2f97e4fe976.mov

## Test plan



* Create a new notebook
* Open share dialog and change the share dropdown from 'Private' to an org, do not check the "Everyone can view..." checkbox, click done to apply the changes
* Go to the notebooks list page, click on the org tab and you should see the previously created notebook
* Login with a separate user that is also a member of the org
  * This user should be able to view & edit the created notebook, as well as view it in the org's notebooks list
* Login with a separate user that is *not* a member of the org
  * You should not be able to access the notebook, or view it in the explore list
* (In dotcom mode) Using an unauthenticated user, navigate to the created notebook
  * You should not be able to access the notebook, or view it in the explore list
* Go to the created notebook share dialog and check the "Everyone can view..." checkbox
* (In dotcom mode) Again, using an unauthenticated user, navigate to the notebook
  * You **should** be able to access the notebook

All read & write permissions are automatically tested at the GQL API layer and at the DB layer.
